internal/handlers/handler_verify.go

```diff
@@ -235,66 +253,174 @@ func updateActivityTimestamp(ctx *middlewares.AutheliaCtx, isBasicAuth bool, use
+	}
 
-	if err != nil {
-		ctx.Error(fmt.Errorf("Unable to parse target URL: %s", err), operationFailedMessage)
-		return
+	// Check Emails.
+	var emailsDelta []string
+	if len(emailsAdded) != 0 {
+		emailsDelta = append(emailsDelta, fmt.Sprintf("Added: %s.", strings.Join(emailsAdded, ", ")))
+	}
+	if len(emailsRemoved) != 0 {
+		emailsDelta = append(emailsDelta, fmt.Sprintf("Removed: %s.", strings.Join(emailsRemoved, ", ")))
 	}
+	if len(emailsDelta) != 0 {
+		// TODO: FOR 4.16.1 RELEASE change Debugf to Tracef
+		ctx.Logger.Debugf("Updated emails detected for %s. %s", userSession.Username, strings.Join(emailsDelta, " "))
+	} else {
+		// TODO: FOR 4.16.1 RELEASE change Debugf to Tracef
+		ctx.Logger.Debugf("No updated emails detected for %s", userSession.Username)
+	}
+}
 
-	if !isSchemeHTTPS(targetURL) && !isSchemeWSS(targetURL) {
-		ctx.Logger.Error(fmt.Errorf("Scheme of target URL %s must be secure since cookies are "+
-			"only transported over a secure connection for security reasons", targetURL.String()))
-		ctx.ReplyUnauthorized()
-		return
+func verifySessionHasUpToDateProfile(ctx *middlewares.AutheliaCtx, targetURL *url.URL, userSession *session.UserSession,
+	refreshProfile bool, refreshProfileInterval time.Duration) error {
+	// TODO: Add a check for LDAP password changes based on a time format attribute.
+	// See https://docs.authelia.com/security/threat-model.html#potential-future-guarantees
+
+	ctx.Logger.Tracef("Checking if we need check the authentication backend for an updated profile for %s.", userSession.Username)
+	if refreshProfile && userSession.Username != "" && targetURL != nil &&
+		ctx.Providers.Authorizer.IsURLMatchingRuleWithGroupSubjects(*targetURL) &&
+		(refreshProfileInterval == schema.RefreshIntervalAlways || userSession.RefreshTTL.Before(ctx.Clock.Now())) {
+		ctx.Logger.Debugf("Checking the authentication backend for an updated profile for user %s", userSession.Username)
+		details, err := ctx.Providers.UserProvider.GetDetails(userSession.Username)
+		// Only update the session if we could get the new details.
+		if err != nil {
+			return err
+		}
+
+		groupsDiff := utils.IsStringSlicesDifferent(userSession.Groups, details.Groups)
+		emailsDiff := utils.IsStringSlicesDifferent(userSession.Emails, details.Emails)
+		if !groupsDiff && !emailsDiff {
+			// TODO: FOR 4.16.1 RELEASE change Debugf to Tracef
+			ctx.Logger.Debugf("Updated profile not detected for %s.", userSession.Username)
+		} else {
+			ctx.Logger.Debugf("Updated profile detected for %s.", userSession.Username)
+			// TODO: FOR 4.16.1 RELEASE: change "debug" to "trace"
+			if ctx.Configuration.LogLevel == "debug" {
+				generateVerifySessionHasUpToDateProfileTraceLogs(ctx, userSession, details)
+			}
+			userSession.Groups = details.Groups
+			userSession.Emails = details.Emails
+
+			// Only update TTL if the user has a interval set.
+			if refreshProfileInterval != schema.RefreshIntervalAlways {
```

Comment:
I think you can remove this block since it can be done by the block at line 335.

Reply:
I actually can't but there was a bug which I fixed. The save session should have been outside that. Providing the two instances of updating. I can explain the reason if necessary.